kimchi/src/circuits/gate.rs

```diff
@@ -225,35 +238,47 @@ impl<F: FftField> LookupInfo<F> {
 
     /// Each entry in `kinds` has a corresponding selector polynomial that controls whether that
     /// lookup kind should be enforced at a given row. This computes those selector polynomials.
-    pub fn selector_polynomials(
+    #[allow(clippy::type_complexity)]
+    pub fn selector_polynomials_and_tables(
         &self,
         domain: &EvaluationDomains<F>,
         gates: &[CircuitGate<F>],
-    ) -> Vec<E<F, D<F>>> {
+    ) -> (Vec<E<F, D<F>>>, Vec<Vec<Vec<F>>>) {
         let n = domain.d1.size as usize;
-        let mut res: Vec<_> = self.kinds.iter().map(|_| vec![F::zero(); n]).collect();
+        let mut selector_values: Vec<_> = self.kinds.iter().map(|_| vec![F::zero(); n]).collect();
+        let mut gate_tables = HashSet::new();
 
         // TODO: is take(n) useful here? I don't see why we need this
         for (i, gate) in gates.iter().enumerate().take(n) {
             let typ = gate.typ;
 
-            if let Some(v) = self.kinds_map.get(&(typ, CurrOrNext::Curr)) {
-                res[*v][i] = F::one();
+            if let Some(selector_index) = self.kinds_map.get(&(typ, CurrOrNext::Curr)) {
+                selector_values[*selector_index][i] = F::one();
```

Comment:
I’m not a fan of this selector index, but confused enough about the flow that I wouldn’t be able to suggest a better way

Reply:
My ultimate plan is for them to go away entirely. IMO we should just be using the gate selectors (added together, under the assumption that they're mutually exclusive) rather than creating a dedicated selector.